# Тести для малих вибірок та пропорцій



**Завдання 1**. E-commerce компанія після редизайну сайту підозрює, що **середній час до покупки (time-to-purchase)** користувачів **збільшився**.
Історично середній час від першого заходу на сайт до покупки становив $\mu_0 = 123.7$ хвилин.

Команда аналітиків випадково обрала дані **7 покупців після редизайну**:
`128, 135, 121, 142, 126, 151, 123`

З допомогою наявних даних зʼясуйте, чи збільшився середній час покупки після редизайну.

Для цього виконайте наступні 6 кроків. Правильне виконання кожного з кроків оцінюється в 1 бал.

1. Запишіть нульову та альтернативну гіпотези і визначте тип тесту.
2. Обчисліть вибіркові статистики: $\bar x$, $s$, $n$.
3. Оберіть тип тесту та виконайте його (знайдіть критичне значення тесту, статистику тесту та р-значення) будь-яким пасуючим способом, щоб перевірити гіпотезу на рівні значущості $\alpha = 0.10$.
4. Прийміть рішення, чи відхиляєте ви гіпотезу $H_0$ використовуючи p-value.
5. Напишіть висновок: чи справді редизайн сайту подовжив час до покупки?
6. Чи зміниться ваше рішення при зміні рівня значущості на $\alpha = 0.05$.


In [1]:
import numpy as np
from scipy import stats

In [5]:
# дані
data = np.array([128, 135, 121, 142, 126, 151, 123])

mu0 = 123.7
alpha = 0.05
n = len(data)

print("Тип тесту: правосторонній t-test (мала вибірка, σ невідоме)")

# обчислення
x_bar = np.mean(data)
s = np.std(data, ddof=1)

t_stat = (x_bar - mu0) / (s / np.sqrt(n))

df = n-1
t_crit = stats.t.ppf(1-alpha, df)

p_value = 1 - stats.t.cdf(t_stat, df)

print(f"Вибіркове середнє: {x_bar:.2f}")
print(f"Вибіркове стандартне відхилення: {s:.2f}")
print(f"t-статистика: {t_stat:.3f}")
print(f"Критичне значення (df={df}): {t_crit:.3f}")
print(f"p-value: {p_value:.4f}")

# висновок
if t_stat > t_crit:
    print(f"Відхиляємо H0 на рівні значущості {alpha}")
else:
    print("Не відхиляємо H0: доказів недостатньо")

Тип тесту: правосторонній t-test (мала вибірка, σ невідоме)
Вибіркове середнє: 132.29
Вибіркове стандартне відхилення: 10.98
t-статистика: 2.069
Критичне значення (df=6): 1.943
p-value: 0.0420
Відхиляємо H0 на рівні значущості 0.05


Висновок:
Є статистично значущі підстави вважати, що середній час до покупки збільшився після редизайну.

**Завдання 2.**

До спеціальної рекламної кампанії **23%** дорослих упізнавали логотип компанії. Після завершення кампанії відділ маркетингу провів опитування: з **1200** випадково відібраних дорослих **311** упізнали логотип.

Перевірте на рівні значущості **3%** ($\alpha=0.03$), чи дають ці дані достатні підстави стверджувати, що **тепер більше ніж 23%** дорослих упізнають логотип компанії. Для розвʼязку використовуйте бібліотеку `statsmodels`.

Зробіть висновок, чи зросла впізнаваність логотипу.

Додатково, обчисліть довірчий інтревал на заданому рівні значущості і проінтерпретуйте текстом - як він додатково пояснює прийняте нами рішення?

In [6]:
from statsmodels.stats.proportion import proportions_ztest, proportion_confint

In [7]:
# дані
count = 311
nobs = 1200
p0 = 0.23
alpha = 0.03

print("Тип тесту: правосторонній Z-test для однієї пропорції")

# гіпотези
print("H0: p = 0.23")
print("Ha: p > 0.23")

# оцінка пропорції
p_hat = count / nobs

# z-test
z_stat, p_value = proportions_ztest(count,
                                    nobs,
                                    value=p0,
                                    alternative='larger')

# довірчий інтервал
confint = proportion_confint(count,
                             nobs,
                             alpha=0.05,
                             method='normal')

print(f"Вибіркова пропорція: {p_hat:.3f}")
print(f"Z-статистика: {z_stat:.3f}")
print(f"p-value: {p_value:.4f}")
print(f"95% довірчий інтервал: ({confint[0]:.3f}, {confint[1]:.3f})")

# висновок
if p_value < alpha:
    print(f"Відхиляємо H0 на рівні значущості {alpha}")
    print("Є статистично значущі підстави вважати, що пропорція збільшилась")
else:
    print("Не відхиляємо H0: доказів недостатньо")

Тип тесту: правосторонній Z-test для однієї пропорції
H0: p = 0.23
Ha: p > 0.23
Вибіркова пропорція: 0.259
Z-статистика: 2.306
p-value: 0.0106
95% довірчий інтервал: (0.234, 0.284)
Відхиляємо H0 на рівні значущості 0.03
Є статистично значущі підстави вважати, що пропорція збільшилась


Висновок:
Оскільки p-value = 0.0106 < α = 0.03, ми відхиляємо нульову гіпотезу. Є статистично значущі підстави вважати, що впізнаваність логотипу зросла після рекламної кампанії.

Інтерпретація довірчого інтервалу:
95% довірчий інтервал для пропорції становить приблизно (0.234; 0.284), що означає, що справжня частка користувачів, які впізнають логотип, з імовірністю 95% знаходиться в цьому діапазоні.

Пояснення рішення:
Оскільки більша частина довірчого інтервалу лежить вище 0.23, це додатково підтверджує висновок про статистично значуще зростання впізнаваності логотипу.